Add SSDT sqlproj specific gitignore file for database development usi…

sqltoolbox opened the pull request:
…ng t-sql.

**Reasons for making this change:**
<!-- Include your relationship to the project and what you expect to get from this change. -->

I am a database developer working with database projects that are source controlled. I've noticed there isn't a sqlproj t-sql specific gitignore file. I will amend this template to make it specific to SSDT sqlproj projects/solutions.

**Links to documentation supporting these rule changes:**

https://learn.microsoft.com/en-us/sql/ssdt/sql-server-data-tools?view=sql-server-ver16
https://learn.microsoft.com/en-us/sql/ssdt/database-project-settings?view=sql-server-ver16

If this is a new template:

 - **Link to application or project’s homepage**: 
 https://learn.microsoft.com/en-us/sql/ssdt/sql-server-data-tools?view=sql-server-ver16
 https://learn.microsoft.com/en-us/sql/ssdt/database-project-settings?view=sql-server-ver16

thepuzzeler committed:
Add SSDT sqlproj specific gitignore file for database development using t-sql.

wirecat commented:
@sqltoolbox This template does seem very similar to the Visual Studio template 🤔 
Would it be possible to list the SSDT specific rules?

My first inclination is to add the very specific SSDT rules under our `community/` folder, but I'd like to understand the needs a bit better first.

sqltoolbox committed:
SSDT specific objects ignore

A trimmed down version of the gitignore file tailored specifically towards SSDT projects in Visual Studio.
